client/clientutil/modelinfo.go

```diff
@@ -0,0 +1,461 @@
+// YAML format. The output will be written to the provided io.Writer.
+func PrintSerialAssertionYAML(w *tabwriter.Writer, modelFormatter ModelFormatter, options PrintModelAssertionOptions, serialAssertion asserts.Serial) error {
+
+	// if we got an invalid model assertion bail early
+	if options.Assertion {
+		_, err := w.Write(asserts.Encode(&serialAssertion))
+		return err
+	}
+
+	// the rest of this function is the main flow for outputting either the
+	// serial assertion in normal or verbose mode
+
+	// ordering of the primary keys for model: brand, model, serial
+	// ordering of primary keys for serial is brand-id, model, serial
+
+	// output brand/brand-id
+	brandIDHeader := serialAssertion.HeaderString("brand-id")
+	modelHeader := serialAssertion.HeaderString("model")
+	serial := serialAssertion.HeaderString("serial")
+
+	fmt.Fprintf(w, "brand-id:\t%s\n", brandIDHeader)
+	fmt.Fprintf(w, "model:\t%s\n", modelHeader)
+	fmt.Fprintf(w, "serial:\t%s\n", serial)
+
+	// return if verbose is not requested
+	if !options.Verbose {
+		return w.Flush()
+	}
+	return printVerboseAssertionHeaders(w, options, &serialAssertion)
+}
+
+// PrintModelAssertionJSON will format the provided serial or model assertion based on the parameters given in
+// JSON format. The output will be written to the provided io.Writer.
+func PrintModelAssertionJSON(w *tabwriter.Writer, modelFormatter ModelFormatter, options PrintModelAssertionOptions, modelAssertion asserts.Model, serialAssertion *asserts.Serial) error {
+
+	serializeJSON := func(v interface{}) error {
+		marshalled, err := json.MarshalIndent(v, "", "  ")
+		if err != nil {
+			return err
+		}
+		_, err = w.Write(marshalled)
+		w.Flush()
+		return err
+	}
+
+	if options.Assertion {
+		modelJSON := modelAssertJSON{}
+		modelJSON.Headers = modelAssertion.Headers()
+		modelJSON.Body = string(modelAssertion.Body())
+		return serializeJSON(modelJSON)
+	}
+
+	// ordering of the primary keys for model: brand, model, serial
+	// ordering of primary keys for serial is brand-id, model, serial
+
+	// output brand/brand-id
+	modelData := make(map[string]interface{})
+	modelData["brand-id"] = modelAssertion.HeaderString("brand-id")
+	modelData["model"] = modelAssertion.HeaderString("model")
+
+	// for the serial header, if there's no serial yet, it's not an error for
+	// model (and we already handled the serial error above) but need to add a
+	// parenthetical about the device not being registered yet
+	var serial string
+	if serialAssertion == nil {
+		// For JSON verbose is always true, so we need to escape the dash
+		serial = modelFormatter.GetEscapedDash()
+		serial += " (device not registered yet)"
+	} else {
+		serial = serialAssertion.HeaderString("serial")
+	}
+
+	// only output the grade if it is non-empty, either it is not in the model
+	// assertion for all non-uc20 model assertions, or it is non-empty and
+	// required for uc20 model assertions
```

Comment:
Same remark as for the other identical comment

Reply:
Fixed!